### PR TITLE
Fixes tray harvest update and weed age

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

When a non-repeat harvest plant was harvested, it did not update the `lastproduce` variable, resulting in the next plant that was planted in the same tray having to wait until being at least the same age as the previous plant before being able to be harvested. To fix this, harvesting a non-repeat harvest plant now always sets `lastproduce` to 0.

Weeds overtaking a tray mistakenly started at age 0, unlike regular planted seeds that started at age 1. Weeds now also start at age 1.

## Why It's Good For The Game

Less bugs.

## Changelog
:cl: githubuser343
fix: Trays now properly cleanup after non-repeat harvest plants were harvested.
fix: Weeds overtaking a tray start at same age as freshly planted seeds. 
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/hydroponics.dm [hunk 1]
```diff
@@ -380,7 +380,8 @@
 			myseed = new /obj/item/seeds/plump(src)
 		else
 			myseed = new /obj/item/seeds/starthistle(src)
-	age = 0
+	age = 1
+	lastproduce = 0
```
Comment (park66665) [inline code/modules/hydroponics/hydroponics.dm]:
```suggestion
	lastproduce = 1
```

File: code/modules/hydroponics/hydroponics.dm [hunk 2]
```diff
@@ -786,6 +787,7 @@
 			myseed = O
 			update_name()
 			age = 1
+			lastproduce = 0
```
Comment (park66665) [inline code/modules/hydroponics/hydroponics.dm]:
```suggestion
			lastproduce = 1
```

**Review by park66665 — COMMENTED** [review]
`lastproduce` is set to *follow* the plant *age* as it grows up. As the `age` starts as 1, `lastproduce`, then, has to start as 1 too.

Also, to note, most of the cases it doesn't matter very much since plant growth makes `lastproduce` follow the `age`. However there *are* cases where non-cleaned `lastproduce` causing problem due to plants maturation time being 2 or less, for maturation time less or equal to 2 skips the `lastproduce`'s following of `age` of maturation period: last plants arbitrary `lastproduce` is then left uncleaned, haunting the future plants harvest.

**Comment by ghost** [comment]
That's not how it works, but have it as you wish.

**Comment by park66665** [comment]
> That's not how it works, but have it as you wish.

https://github.com/BeeStation/BeeStation-Hornet/blob/917fbb231ccef50e1384c90b44b34ed6b0166c15/code/modules/hydroponics/hydroponics.dm#L104-L125

Take a closer look. Especially that `lastproduce = age` part.

Merging a PR with cursory perusal depends on trust; especially, trust by maintainers of contributors. You *just* have undermined that trust by your last comment. You now should explain what were you thinking while making this fix with precise locations of code lines involved if you want this PR merged by me.

**Comment by ghost** [comment]
What trust? Because someone accuses me of being an alt account of someone else? I frankly do not care if you want bugs fixed or how you justify rejecting PRs based on hearsay.

I code out of my free time to better the game. You're free to fix it yourself, I've got better things to do.

**Comment by park66665** [comment]
> What trust? Because someone accuses me of being an alt account of someone else? I frankly do not care if you want bugs fixed or how you justify rejecting PRs based on hearsay.

You are missing the point -- greatly. The 'trust' here is me knowing that you know what you are doing. However, this doesn't matter as of now.

> I code out of my free time to better the game. You're free to fix it yourself, I've got better things to do.

This is completely out of point. You simply are justifying your act of leaving this project. I have nothing to say since I also work for free -- and review and merge PRs I don't like very much.

Anyways, good luck to whatever "better things to do" you've gotten.

**Review by park66665 — APPROVED** [review]
Okay, Let me give this PR a go.

All clear. This PR is good to go.

**Pull request merged**